### PR TITLE
SONAR-7984 WS api/server_id/generate

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/serverid/ws/GenerateAction.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      .setHandler(this);
+
+    action.createParam(PARAM_ORGANIZATION)
+      .setDescription("Organization name")
+      .setExampleValue("SonarSource")
+      .setRequired(true);
+
+    action.createParam(PARAM_IP)
+      .setDescription("IP address")
+      .setExampleValue("10.142.20.56")
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/serverid/ws/GenerateAction.java]: MINOR Make this IP "10.142.20.56" address configurable. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1313)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [serverid](https://github.com/SonarSource/sonarqube/blob/8f6ccbc520fba4a18d414a4f02671deb5f1b5b62/sonar-ws/src/main/java/org/sonarqube/ws/client/serverid): Add a 'package-info.java' file to document the 'serverid' package [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: server/sonar-server/src/main/java/org/sonar/server/serverid/ws/GenerateAction.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import org.sonarqube.ws.client.serverid.GenerateRequest;
+
+import static org.sonar.api.CoreProperties.ORGANISATION;
+import static org.sonar.api.CoreProperties.PERMANENT_SERVER_ID;
+import static org.sonar.api.CoreProperties.SERVER_ID_IP_ADDRESS;
+import static org.sonar.core.permission.GlobalPermissions.SYSTEM_ADMIN;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+
+public class GenerateAction implements ServerIdWsAction {
+  public static final String PARAM_ORGANIZATION = "organization";
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/serverid/ws/GenerateAction.java]: As the ui is displaying "organisation", I would suggest to also use this and not "organization" everywhere.

File: server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdGenerator.java [hunk 3]
```diff
@@ -70,7 +72,20 @@ public String generate(String organisationName, String ipAddress) {
     return id;
   }
 
-  boolean isValidOrganizationName(String organisation) {
+  public String generateOrFail(String organizationName, String ipAddress) {
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/platform/ServerIdGenerator.java]: I don't understand this method : there's already a generate() method.
The code written should be moved to the ws action.

File: server/sonar-server/src/main/resources/org/sonar/server/serverid/ws/generate-example.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "serverId": "server_id"
```
Comment (julienlancelot) [inline server/sonar-server/src/main/resources/org/sonar/server/serverid/ws/generate-example.json]: please replace this value by a random one (1818a1eefb26f9g for instance)

**Comment by julienlancelot** [comment]
Let's talk about this generateOrFail method

**Comment by julienlancelot** [comment]
LGTM after applying decision taken during offline meeting

File: server/sonar-server/src/main/java/org/sonar/server/serverid/ws/GenerateAction.java (hunk 2 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/serverid/ws/GenerateAction.java]: I check with the documentation team. I'll go with the s meanwhile and create a ticket if needed

Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/serverid/ws/GenerateAction.java]: I retract: https://github.com/SonarSource/sonarqube/blob/bae4f619db5198f7cb67a7848fdff756d83b5744/sonar-core/src/main/resources/org/sonar/l10n/core.properties#L2078-L2078

File: server/sonar-server/src/main/resources/org/sonar/server/serverid/ws/generate-example.json (hunk 4 above)
Comment (teryk) [inline server/sonar-server/src/main/resources/org/sonar/server/serverid/ws/generate-example.json]: I don't want to be teasing but that's you gave in the show-example.json 😄

Comment (julienlancelot) [inline server/sonar-server/src/main/resources/org/sonar/server/serverid/ws/generate-example.json]: :( Then please also update mine !

**Pull request merged**